File: okex-php-sdk-api-v5/src/FundingApi.php
<?php


namespace okv5;


class FundingApi extends Utils
{
//
    public function getDepositAddress($ccy)
    {
        $params = [
            'ccy' => $ccy,
        ];

        return $this->request('/api/v5/asset/deposit-address',$params, 'GET');
    }

    //
    public function getBalance($ccy='')
    {
        $params = [
            'ccy' => $ccy,
        ];
        return $this->request('/api/v5/asset/balances',$params, 'GET');
    }

    //
    public function transfer($ccy,$amt,$type='',$from,$to,$subAcct='',$instId='',$toInstId='',$loanTrans='')
    {
        $params = [
            'ccy' => $ccy,
            'amt' => $amt,
            'type' => $type,
            'from' => $from,
            'to' => $to,
            'subAcct' => $subAcct,
            'instId' => $instId,
            'toInstId' => $toInstId,
            'loanTrans' => $loanTrans,
        ];

        return $this->request('/api/v5/asset/transfer',$params, 'POST');
    }

    public function transferState($transId,$type='')
    {
        $params = [
            'transId' => $transId,
            'type' => $type,
        ];

        return $this->request('/api/v5/asset/transfer-state',$params, 'GET');
    }

    public function withdrawal($ccy,$chain='',$amt,$dest,$toAddr,$fee)
    {
        $params = [
            'ccy' => $ccy,
            'chain' => $chain,
            'amt' => $amt,
            'dest' => $dest,
            'toAddr' => $toAddr,
            'fee' => $fee,
        ];

        return $this->request('/api/v5/asset/withdrawal',$params, 'POST');
    }

    public function getDepositHistory($ccy='',$depId='',$state='',$after='',$before='',$limit='',$txId='')
    {
        $params = [
            'ccy' => $ccy,
            'depId' => $depId,
            'state' => $state,
            'after' => $after,
            'before' => $before,
            'limit' => $limit,
            'txId' => $txId,
        ];

        return $this->request('/api/v5/asset/deposit-history',$params, 'GET');
    }

    public function getWithdrawalHistory($ccy='',$wdId='',$clientId='',$state='',$after='',$before='',$limit='',$txId='')
    {
        $params = [
            'ccy' => $ccy,
            'wdId' => $wdId,
            'clientId' => $clientId,
            'state' => $state,
            'after' => $after,
            'before' => $before,
            'limit' => $limit,
            'txId' => $txId,
        ];

        return $this->request('/api/v5/asset/withdrawal-history',$params, 'GET');
    }

    public function convertDustAssets($ccy)
    {
        $params = [
            'ccy' => $ccy,
        ];

        return $this->request('/api/v5/asset/convert-dust-assets',$params, 'POST');
    }

    public function getCurrencies()
    {
        return $this->request('/api/v5/asset/currencies','', 'GET');
    }

    public function PurchaseRedempt($ccy,$amt,$side,$rate)
    {
        $params = [
            'ccy' => $ccy,
            'amt' => $amt,
            'side' => $side,
            'rate' => $rate,
        ];

        return $this->request('/api/v5/asset/purchase_redempt',$params, 'POST');
    }

    public function getBills($ccy='',$type='',$after='',$before='',$limit='')
    {
        $params = [
            'ccy' => $ccy,
            'type' => $type,
            'after' => $after,
            'before' => $before,
            'limit' => $limit,
        ];

        return $this->request('/api/v5/asset/bills',$params, 'GET');
    }

    public function getPiggyBalance($ccy='')
    {
        $params = [
            'ccy' => $ccy,
        ];

        return $this->request('/api/v5/asset/saving-balance',$params, 'GET');
    }

    public function depositLightning($ccy,$amt,$to='')
    {
        $params = [
            'ccy' => $ccy,
            'amt' => $amt,
            'to' => $to,
        ];

        return $this->request('/api/v5/asset/deposit-lightning',$params, 'GET');
    }

    public function withdrawalLightning($ccy,$invoice)
    {
        $params = [
            'ccy' => $ccy,
            'invoice' => $invoice,
        ];

        return $this->request('/api/v5/asset/withdrawal-lightning',$params, 'POST');
    }

    public function cancelWithdrawal($wdId)
    {
        $params = [
            'wdId' => $wdId,
        ];

        return $this->request('/api/v5/asset/cancel-withdrawal',$params, 'POST');
    }

    public function assetValuation($ccy='')
    {
        $params = [
            'ccy' => $ccy,
        ];

        return $this->request('/api/v5/asset/asset-valuation',$params, 'GET');
    }

    public function setLendingRate($ccy,$rate)
    {
        $params = [
            'ccy' => $ccy,
            'rate' => $rate,
        ];

        return $this->request('/api/v5/asset/set-lending-rate',$params, 'POST');
    }

    public function lendingHistory($ccy='',$after='',$before='',$limit='')
    {
        $params = [
            'ccy' => $ccy,
            'after' => $after,
            'before' => $before,
            'limit' => $limit,
        ];

        return $this->request('/api/v5/asset/lending-history',$params, 'GET');
    }

    public function lendingRateSummary($ccy='')
    {
        $params = [
            'ccy' => $ccy,
        ];

        return $this->request('/api/v5/asset/lending-rate-summary',$params, 'GET');
    }

    public function lendingRateHistory($ccy='',$after='',$before='',$limit='')
    {
        $params = [
            'ccy' => $ccy,
            'after' => $after,
            'before' => $before,
            'limit' => $limit,
        ];

        return $this->request('/api/v5/asset/lending-rate-history',$params, 'GET');
    }

    public function positionsHistory($instType='',$instId='',$mgnMode='',$type='',$after='',$before='',$limit='')
    {
        $params = [
            'instType' => $instType,
            'instId' => $instId,
            'mgnMode' => $mgnMode,
            'type' => $type,
            'after' => $after,
            'before' => $before,
            'limit' => $limit,
        ];

        return $this->request('/api/v5/account/positions-history',$params, 'GET');
    }

    public function nonTradableAssets($ccy='')
    {
        $params = [
            'ccy' => $ccy,
        ];

        return $this->request('/api/v5/asset/non-tradable-assets',$params, 'GET');
    }

    public function depositWithdrawStatus($wdId='',$txId='',$ccy='',$to='',$chain='')
    {
        $params = [
            'wdId' => $wdId,
            'txId' => $txId,
            'ccy' => $ccy,
            'to' => $to,
            'chain' => $chain,
        ];

        return $this->request('/api/v5/asset/deposit-withdraw-status',$params, 'GET');
    }

    public function exchangeList()
    {
        
        return $this->request('/api/v5/asset/exchange-list','', 'GET');
    }

    public function getMonthlyStatement($month='')
    {
        $params = [
            'month' => $month,
        ];

        return $this->request('/api/v5/asset/monthly-statement',$params, 'GET');
    }

    public function currencies()
    {
        
        return $this->request('/api/v5/asset/convert/currencies','', 'GET');
    }

    public function currencyPair($fromCcy,$toCcy)
    {
        $params = [
            'fromCcy' => $fromCcy,
            'toCcy' => $toCcy,
        ];

        return $this->request('/api/v5/asset/convert/currency-pair',$params, 'GET');
    }
    
    public function estimateQuote($baseCcy,$quoteCcy,$side,$rfqSz,$rfqSzCcy,$clQReqId='',$tag='')
    {
        $params = [
            'baseCcy' => $baseCcy,
            'quoteCcy' => $quoteCcy,
            'side' => $side,
            'rfqSz' => $rfqSz,
            'rfqSzCcy' => $rfqSzCcy,
            'clQReqId' => $clQReqId,
            'tag' => $tag,
        ];

        return $this->request('/api/v5/asset/convert/estimate-quote',$params, 'POST');
    }

    public function trade($quoteId,$baseCcy,$quoteCcy,$side,$sz,$szCcy,$clQReqId='',$tag='')
    {
        $params = [
            'quoteId' => $quoteId,
            'baseCcy' => $baseCcy,
            'quoteCcy' => $quoteCcy,
            'side' => $side,
            'sz' => $sz,
            'szCcy' => $szCcy,
            'clQReqId' => $clQReqId,
            'tag' => $tag,
        ];

        return $this->request('/api/v5/asset/convert/trade',$params, 'POST');
    }

    public function history($clTReqId='',$after='',$before='',$limit='',$tag='')
    {
        $params = [
            'clTReqId' => $clTReqId,
            'after' => $after,
            'before' => $before,
            'limit' => $limit,
            'tag' => $tag,
        ];

        return $this->request('/api/v5/asset/convert/history',$params, 'GET');
    }

    public function billsHistory($ccy='',$type='',$clientId='',$after='',$before='',$limit='',$pagingType='')
    {
        $params = [
            'ccy' => $ccy,
            'type' => $type,
            'clientId' => $clientId,
            'after' => $after,
            'before' => $before,
            'limit' => $limit,
            'pagingType' => $pagingType,
        ];

        return $this->request('/api/v5/asset/bills-history',$params, 'GET');
    }

}

File: okex-java-sdk-api-v5/src/main/java/com/okex/open/api/service/funding/impl/FundingAPI.java
package com.okex.open.api.service.funding.impl;

import com.alibaba.fastjson.JSONObject;
import retrofit2.Call;
import retrofit2.http.*;


interface FundingAPI {

    //获取币种列表 Get Currencies
    @GET("/api/v5/asset/currencies")
    Call<JSONObject> getCurrencies();

    //获取资金账户余额
    @GET("/api/v5/asset/balances")
    Call<JSONObject> getBalance(@Query("ccy") String ccy);

    //资金划转  Funds Transfer
    @POST("/api/v5/asset/transfer")
    Call<JSONObject> fundsTransfer(@Body JSONObject jsonObject);

    //资金流水查询 Asset Bills Details
    @GET("/api/v5/asset/bills")
    Call<JSONObject> assetBillsDetails(@Query("ccy")String ccy,@Query("type")String type,@Query("clientId")String clientId,@Query("after")String after,@Query("before")String before,@Query("limit")String limit);

    //获取充值地址信息 Get Deposit Address
    @GET("/api/v5/asset/deposit-address")
    Call<JSONObject> getDepositAddress(@Query("ccy") String ccy);

    //获取充值记录 Get Deposit History
    @GET("/api/v5/asset/deposit-history")
    Call<JSONObject> getDepositHistory(@Query("ccy") String ccy,@Query("txId") String txId,@Query("state")String state,@Query("after")String after,@Query("before")String before,@Query("limit")String limit,@Query("depId")String depId);

    //提币 Withdrawal
    @POST("/api/v5/asset/withdrawal")
    Call<JSONObject> Withdrawal(@Body JSONObject jsonObject);

    //获取提币记录 Get Withdrawal History
    @GET("/api/v5/asset/withdrawal-history")
    Call<JSONObject> getWithdrawalHistory(@Query("ccy") String ccy,@Query("clientId") String clientId,@Query("txId") String txId,@Query("state")String state,@Query("after")String after,@Query("before")String before,@Query("limit")String limit,@Query("wdId")String wdId);

    //余币宝申购/赎回 PiggyBank Purchase/Redemption
    @POST("/api/v5/asset/purchase_redempt")
    Call<JSONObject> piggyBankPurchaseRedemption(@Body JSONObject jsonObject);

    //获取余币宝余额
    @GET("/api/v5/asset/saving-balance")
    Call<JSONObject> piggyBalance(@Query("ccy") String ccy);

    //闪电网络充币 Deposit Lightning
    @GET("/api/v5/asset/deposit-lightning")
    Call<JSONObject> depositLightning(@Query("ccy")String ccy, @Query("amt")String amt, @Query("to")String to);

    //闪电网络提币 Withdrawal Lightning
    @POST("/api/v5/asset/withdrawal-lightning")
    Call<JSONObject>  withdrawalLightning(@Body JSONObject jsonObject);

    //获取资金划转状态 Transfer State
    @GET("/api/v5/asset/transfer-state")
    Call<JSONObject> transferState(@Query("transId")String transId, @Query("clientId")String clientId ,@Query("type")String type);

    //获取账户资产估值 Get account asset valuation
    @GET("/api/v5/asset/asset-valuation")
    Call<JSONObject> assetValuation(@Query("ccy") String ccy);

    //设置余币宝借贷利率 Set lending rate
    @POST("/api/v5/asset/set-lending-rate")
    Call<JSONObject>  SetLendingRate(@Body JSONObject jsonObject);

    //获取余币宝出借明细  Get lending history
    @GET("/api/v5/asset/lending-history")
    Call<JSONObject> lendingHistory(@Query("ccy") String ccy,@Query("after") String after,@Query("before") String before,@Query("limit") String limit);

    //获取市场借贷信息（公共） Get public borrow info (public)
    @GET("/api/v5/asset/lending-rate-summary")
    Call<JSONObject> lendingRateSummary(@Query("ccy") String ccy);

    //获取市场借贷历史（公共）  Get public borrow history (public)
    @GET("/api/v5/asset/lending-rate-history")
    Call<JSONObject> lendingRateHistory(@Query("ccy") String ccy,@Query("after") String after,@Query("before") String before,@Query("limit") String limit);

    @POST("/api/v5/asset/convert-dust-assets")
    Call<JSONObject> ConvertDustAssets(@Body JSONObject parseObject);

    @POST("/api/v5/asset/cancel-withdrawal")
    Call<JSONObject> cancelWithdrawal(@Body JSONObject parseObject);
}